lib/extended-css-parser.js

```diff
@@ -101,7 +101,7 @@ var ExtendedCssParser = (function() { // jshint ignore:line
 
             while (this.cssText) {
                 // Apply tolerant tokenization.
-                var parseResult = Sizzle.tokenize(this.cssText, false, true);
+                var parseResult = Sizzle.tokenize(this.cssText, false, { tolerant: true, noCache: true });
```

Comment:
Why do you need `noCache` here?

Reply:
The point of applying Sizzle at parse time is to cache the tokenize result.

Reply:
Otherwise, it will return a re-sorted result. This is okay for Sizzle, but browsers do not understand pseudos standing before tag selectors (`:not(smth)div`).

Reply:
Why does that matter?

Reply:
This selector is then divided into the `simple` and `complex` parts and the `simple` part does not work.

Reply:
The way it calls `tokenize` in extended-css-parser.js` has unnecessary overhead. It is also very confusing.
I suggest that we should not use `noCache` and construct a valid css selector from already-shaken tokens array in `extended-css-selector.js`. It will be cheaper than calling `tokenize` one more time.

Reply:
Single tokenize with noCache call is about 0.011 ms on my computer, might be more on yours. On one hand, that's not too much, but on the other -- this is almost as much as it takes Sizzle to exec a simple selector.

Overall: valid point, thank you! I'll need to think about it a bit.

Reply:
Fixed